Source/Csla/IDataPortal.cs

```diff
@@ -35,14 +37,14 @@ public interface IDataPortal
     /// </summary>
     /// <param name="obj">Object to update.</param>
     /// <exception cref="ArgumentNullException"><paramref name="obj"/> is <see langword="null"/>.</exception>
-    Task<object> UpdateAsync(object obj);
+    Task<ICslaObject> UpdateAsync(ICslaObject obj);
     /// <summary>
     /// Called by a factory method in a business class or
     /// by the UI to execute a command object.
     /// </summary>
     /// <param name="command">Command object to execute.</param>
     /// <exception cref="ArgumentNullException"><paramref name="command"/> is <see langword="null"/>.</exception>
-    Task<object> ExecuteAsync(object command);
+    Task<ICslaObject> ExecuteAsync(ICslaObject command);
```

Comment:
Shouldn't we "promote" this to `ICommandObject` instead of `ICslaObject`? It has to be a command to be `Executed` correctly, as far as I understand.

Reply:
We can't do this because of the constrain that `T` needs to be `ICslaObject`